modules/optimization/src/optimizationreporters/ParameterMeshOptimization.C

```diff
@@ -75,8 +74,59 @@ ParameterMeshOptimization::validParams()
 }
 
 ParameterMeshOptimization::ParameterMeshOptimization(const InputParameters & parameters)
-  : OptimizationDataTempl<OptimizationReporterBase>(parameters)
+  : GeneralOptimization(parameters)
 {
+}
+
+std::vector<Real>
+ParameterMeshOptimization::parseData(const std::vector<unsigned int> & exodus_timestep,
+                                     const ParameterMesh & pmesh,
+                                     Real constantDataFromInput,
+                                     const std::string & meshVarName,
+                                     unsigned int ntimes) const
+{
+  unsigned int numberOfControllableParameters = pmesh.size() * ntimes;
+  std::vector<Real> parsedData;
+  // read from mesh
+  if (!meshVarName.empty())
+  {
+    for (auto const & step : exodus_timestep)
+    {
+      std::vector<Real> data = pmesh.getParameterValues(meshVarName, step);
+      parsedData.insert(parsedData.end(), data.begin(), data.end());
+    }
+    if (parsedData.size() != numberOfControllableParameters)
+      mooseError("Number of parameters assigned by ",
+                 meshVarName,
+                 " is not equal to the number of parameters on the mesh.  Mesh contains ",
+                 numberOfControllableParameters,
+                 " parameters and ",
+                 meshVarName,
+                 " assigned ",
+                 parsedData.size(),
+                 " parameters.");
+  }
+  else // read in constant or default values
+    parsedData.resize(parsedData.size() + numberOfControllableParameters, constantDataFromInput);
+
+  return parsedData;
+}
+
+// function only used for test objects
+void
+ParameterMeshOptimization::setSimulationValuesForTesting(std::vector<Real> & /*data*/)
+{
+  // _simulation_values.clear();
+  // _simulation_values = data;
```

Comment:
?

Reply:
Function was eliminated, deleted now. This was a round about way of testing the code that was used before.